### PR TITLE
Improve S2589: Update warning message for null check

### LINKED ISSUE #8010: S2589: Improve message in the case of null propagating operator
### Description

In some cases, the problem is hard to understand and we should make this more visible for the users.

### Repro steps

` _componentPage = (_component?.Type == null) ? _parentComponent?.Name : _component?.Name;`

or

```
if (_component?.Type != null)
  return $"components/{_component?.Link}";
```
image

### Expected behavior

For the given examples, the message should make it clear that the null propagating operator is redundant.

### Actual behavior

The raised message is "Change this expression which always evaluates to the same result." highlighting the identifier, not the operator, which can be confusing.

### PR DESCRIPTION
**Pull request by zsolt-kolbay-sonarsource**
Fixes #8010

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.vb [hunk 1]
```diff
@@ -1711,12 +1711,12 @@ Public Class NullOrEmpty
 
         'Combo/Fatality
         ret = If(isNotNothing, isNothing)
-        '        ^^^^^^^^^^^^                             Noncompliant {{Change this expression which always evaluates to the same result. Some code paths are unreachable.}}
+        '        ^^^^^^^^^^^^                             Noncompliant {{Remove this unnecessary check for null. Some code paths are unreachable.}}
```
Comment (zsolt-kolbay-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.vb:1714]: Not sure about VB.NET: is `check for null` good enough or should it be `check for Nothing`?

Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.vb:1714]: It should be `Nothing`.

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
The functionality was there already, just the message needed improving. That makes this much easier than anticipated! LGTM except for the VB message.

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM!

**Pull request merged**